compiler/qsc_ast/src/mut_visit.rs

```diff
@@ -384,12 +389,26 @@ pub fn walk_path(vis: &mut impl MutVisitor, path: &mut Path) {
     vis.visit_ident(&mut path.name);
 }
 
+pub fn walk_path_kind(vis: &mut impl MutVisitor, path: &mut PathKind) {
+    match path {
+        PathKind::Ok(path) => vis.visit_path(path),
+        PathKind::Err(Some(incomplete_path)) => {
+            vis.visit_span(&mut incomplete_path.span);
+
+            for ref mut ident in &mut incomplete_path.segments {
+                vis.visit_ident(ident);
+            }
+        }
+        PathKind::Err(None) => {}
+    }
+}
+
 pub fn walk_ident(vis: &mut impl MutVisitor, ident: &mut Ident) {
     vis.visit_span(&mut ident.span);
 }
 
-pub fn walk_idents(vis: &mut impl MutVisitor, ident: &mut crate::ast::Idents) {
-    for ref mut ident in &mut *ident.0 {
+pub fn walk_idents(vis: &mut impl MutVisitor, idents: &mut [Ident]) {
```

Comment:
Great question, I went back and forth on this for a while. Ultimately, I decided to keep `visit_idents` to avoid updating the current usages and bloating this PR even further. 

This came up in a conversation with Ian as well, and my response was basically:

> In the AST, many of choices came down to "how do I avoid touching a billion lines of code", and this was one of them 

I think I'll sleep on it and see on Monday if I want to open a separate refactoring PR, or push even more changes to this one...